modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java

```diff
@@ -86,13 +88,29 @@ public void destroy() {
     @Override
     public void addSession(Session session) {
 
-        activeDebugSessions.put(session, createSession(session));
+        if (getTenantDomain(session) != null && !activeDebugSessions.containsKey(getTenantDomain(session))) {
+            activeDebugSessions.put(getTenantDomain(session), createSession(session));
+
```

Comment:
Can we remove this additional space here?

Reply:
Fixed with #12